R/automateDataPreparation.R

```diff
@@ -90,7 +90,8 @@ automateDataPreparation <- function(datList = NULL, inputList, path = NULL,
 		stopifnot(class(datList) == "list")
 		stopifnot(class(inputList) == "list")
 		if(is.null(oldIDs)) {oldIDs <- inputList$savFiles$case.id}
-		if(is.null(oldIDs)) stop("Please specify oldIDs. Case ID in inputList$savFiles$case.id seems to be empty.")
+		if(length(datList) == 1 & length(oldIDs) < 1) {oldIDs <- inputList$newID$value}
+		if(is.null(oldIDs) | any(is.na(oldIDs)) | length(oldIDs) < 1) stop("Please use argument 'oldIDs'. Can't take it from inputList$savFiles$case.id, because at least one case-ID seems to be empty.")
```

Comment:
Nit: This part remains a bit hard to read and when trying to refactor it, I got a little stuck 🤯

Maybe, `length(oldIDs) < 1` could become `is.null(oldIDs)` (but maybe, I oversee something in this test)?

Moreover, I prefer multi-line statements for `if`-blocks - except for very, very small blocks (but always, when they are designed throw an `error()` or a `message()`).

The case of missing `case.id` & missing `value` in `newID` is not part of the message, yet (I don't know if it has to).

Taking all of my proposals into account, maybe this code block could work (at least, the tests were successful; this includes lines 92-94):

```r
  if (is.null(oldIDs)) {
    oldIDs <- inputList$savFiles$case.id
  }
  if (is.null(oldIDs) & length(datList) == 1) {
    oldIDs <- inputList$newID$value
  }
  if (is.null(oldIDs) | any(is.na(oldIDs))) {
    stop("Please use argument 'oldIDs'. Can't take it from inputList$savFiles$case.id, because at least one case-ID seems to be empty, or from inputList$newID$value.")
  }
```

Reply:
> Maybe, `length(oldIDs) < 1` could become `is.null(oldIDs)` (but maybe, I oversee something in this test)?

This is not the same. In the case on T:\ oldIDs becomes "character(0)" which is not caputered by `is.null(oldIDs)` but by `length(oldIDs) < 1`

Reply:
Sorry, your suggestion doesn't appear satisfactory to me, because the "character(0)"- case is not captured, which was the initial problem, I think, I still prefer my solution (maybe formatted to multi-line-if-statements is this preference matters)

Reply:
Ah, I see - then, testing the `length()` totally makes sense to me. However, I could still not reproduce the example on `T:/`.

Then, my proposal with `is.null()` is, of course, not appropriate and a refactorization is not a good option.

The multi-line formatting is optional and my personal preference. This might be something that we could discuss on our code convention meeting. 🤔